crates/ty_python_semantic/src/types/class.rs

```diff
@@ -2635,7 +2644,8 @@ impl<'db> KnownClass {
             | Self::ParamSpecKwargs
             | Self::TypeVarTuple
             | Self::NamedTuple
-            | Self::NewType => matches!(module, KnownModule::Typing | KnownModule::TypingExtensions),
+            | Self::NewType
+            | Self::Field => matches!(module, KnownModule::Typing | KnownModule::TypingExtensions),
```

Comment:
This isn't correct -- the module of `Field` is `KnownModule::Dataclasses`, never `KnownModule::Typing` or `KnownModule::TypingExtensions`. You should add it to the first branch of this `match` instead (the `module == self.canonical_module(db)` branch)

Reply:
Indeed. Apologies for the noise -- I should have made it clear that this was not ready for another review, while I was investigating the test failure.

Reply:
oh, no worries!